src/DETHRACE/pc-win95/win95net.c

```diff
@@ -0,0 +1,676 @@
+            }
+        }
+    }
+    ReceiveHostResponses();
+    if (gNumber_of_hosts <= pIndex) {
+        return 0;
+    }
+    dr_dprintf("PDNetGetNextJoinGame(): Adding game.");
+    memcpy(&pGame->pd_net_info.addr_in, &gJoinable_games[pIndex].addr_in, sizeof(pGame->pd_net_info.addr_in));
+    return 1;
+}
+
+// IDA: void __usercall PDNetDisposeGameDetails(tNet_game_details *pDetails@<EAX>)
+void PDNetDisposeGameDetails(tNet_game_details* pDetails) {
+    LOG_TRACE("(%p)", pDetails);
+    NOT_IMPLEMENTED();
+}
+
+// IDA: int __usercall PDNetHostGame@<EAX>(tNet_game_details *pDetails@<EAX>, char *pHost_name@<EDX>, void **pHost_address@<EBX>)
+int PDNetHostGame(tNet_game_details* pDetails, char* pHost_name, void** pHost_address) {
+    LOG_TRACE("(%p, \"%s\", %p)", pDetails, pHost_name, pHost_address);
+
+    dr_dprintf("PDNetHostGame()");
+    *pHost_address = &gLocal_addr;
+    return 1;
+}
+
+// IDA: int __usercall PDNetJoinGame@<EAX>(tNet_game_details *pDetails@<EAX>, char *pPlayer_name@<EDX>)
+int PDNetJoinGame(tNet_game_details* pDetails, char* pPlayer_name) {
+    LOG_TRACE("(%p, \"%s\")", pDetails, pPlayer_name);
+
+    dr_dprintf("PDNetJoinGame()");
+    return 0;
+}
+
+// IDA: void __usercall PDNetLeaveGame(tNet_game_details *pDetails@<EAX>)
+void PDNetLeaveGame(tNet_game_details* pDetails) {
+    LOG_TRACE("(%p)", pDetails);
+
+    dr_dprintf("PDNetLeaveGame()");
+}
+
+// IDA: void __usercall PDNetHostFinishGame(tNet_game_details *pDetails@<EAX>)
+void PDNetHostFinishGame(tNet_game_details* pDetails) {
+    LOG_TRACE("(%p)", pDetails);
+
+    dr_dprintf("PDNetHostFinishGame()");
+}
+
+// IDA: tU32 __usercall PDNetExtractGameID@<EAX>(tNet_game_details *pDetails@<EAX>)
+tU32 PDNetExtractGameID(tNet_game_details* pDetails) {
+    LOG_TRACE("(%p)", pDetails);
+
+    return ntohs(pDetails->pd_net_info.addr_in.sin_port); // PDGetTotalTime();
+}
+
+// IDA: tPlayer_ID __usercall PDNetExtractPlayerID@<EAX>(tNet_game_details *pDetails@<EAX>)
+tPlayer_ID PDNetExtractPlayerID(tNet_game_details* pDetails) {
+    LOG_TRACE("(%p)", pDetails);
+
+    return ntohs(gLocal_addr_ipx->sin_port);
+}
+
+// IDA: void __usercall PDNetObtainSystemUserName(char *pName@<EAX>, int pMax_length@<EDX>)
+void PDNetObtainSystemUserName(char* pName, int pMax_length) {
+#ifdef _WIN32
+    int size;
+    char buffer[16];
+    BOOL result;
+    uint32_t size;
+    char buffer[16];
+    int result;
+
+#endif
+
+    dr_dprintf("PDNetObtainSystemUserName()\n");
+    // todo
+#if 0
+    size = COUNT_OF(buffer);
+    result = GetComputerNameA(buffer, &size);
+    if (result && size != 0) {
+        strncpy(pName, buffer, pMax_length - 1);
+        pName[pMax_length - 1] = '\0';
+    }
+    while (1) {
+        pName = strpbrk(pName, "_=(){}[]<>!$%^&*/:@~;'#,?\\|`\"");
+        if (*pName == '\0') {
+            break;
+        }
+        *pName = '-';
+    }
+#else
+    strcpy(pName, "Ron Turn");
```

Comment:
Perhaps add a FIXME here? Because this can return system specific information.
Windows has `GetComputerNameA`, Linux has `gethostname()` and macos has probably also something.

Reply:
meant to do this before I PR'd. Done now